Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -471,14 +410,16 @@ private func createResolvedPackages(
                 return false
             })
 
+        let lookupByProductIDs = packageBuilder.package.manifest.toolsVersion >= .vNext || moduleAliasingUsed
```

Comment:
I think we should probably centralize this somewhere since eventually we'll want to replace `.vNext` with a concrete version. Maybe an extension on `Manifest` like `disambiguateByProductIdentifier` that checks the tools-version.